### PR TITLE
WLED Audio

### PR DESCRIPTION
**Pull request by netmindz**

### CODE REVIEW

File: src/App/AppEffects.h [hunk 1]
```diff
@@ -223,39 +224,45 @@ class Frizzles2D:public Effect {
   }
 }; // Frizzles2D
 
+#define NUM_GEQ_CHANNELS 16                                           // number of frequency channels. Don't change !!
```
Comment (netmindz) [inline src/App/AppEffects.h]: Should be getter from wledAudioMod

File: src/App/AppEffects.h [hunk 2]
```diff
@@ -223,39 +224,45 @@ class Frizzles2D:public Effect {
...
+
 class GEQEffect:public Effect {
 public:
+  byte previousBarHeight[1024];
+  uint8_t intensity;
+  uint8_t speed;
+  bool check1;
+  bool check2;
```
Comment (netmindz) [inline src/App/AppEffects.h]: Should be named as per their function in the effect, not based on WLED param names

File: src/App/AppEffects.h [hunk 3]
```diff
@@ -223,39 +224,45 @@ class Frizzles2D:public Effect {
...
-    if (millis() - SEGENV.step >= (256U - SEGMENT.intensity)) {
-      SEGENV.step = millis();
+    if (millis() - step >= (256U - intensity)) {
+      step = millis();
       rippleTime = true;
     }
 
-    if (SEGENV.call == 0) SEGMENT.fill(BLACK);
-    int fadeoutDelay = (256 - SEGMENT.speed) / 64;
-    if ((fadeoutDelay <= 1 ) || ((SEGENV.call % fadeoutDelay) == 0)) SEGMENT.fadeToBlackBy(SEGMENT.speed);
```
Comment (netmindz) [inline src/App/AppEffects.h]: To make porting easier and generally code nicer - should we not just a have  `fadeToBlackBy` method that takes only speed (assuming you are controlling all the current LEDs)

File: src/App/AppEffects.h (hunk 1 above)
Comment (ewoudwijma) [reply src/App/AppEffects.h]: Agree

File: src/App/AppEffects.h (hunk 2 above)
Comment (ewoudwijma) [reply src/App/AppEffects.h]: This is quick and dirty, I was thinking of a function like getControls which calls initSlider(..., control name, ...)

Similar as now done in fx.chFun

This function does not claim room of variables in the class.

Only drawback is that the effect itself should call getValue("control name") which is too slow, so maybe store that value during setup(), but then again we create local class variables...

File: src/App/AppEffects.h (hunk 3 above)
Comment (ewoudwijma) [reply src/App/AppEffects.h]: Maybe, don't know yet, I now converted them to the fastled version, which I like also (the fastled compatibility)

Comment (netmindz) [reply src/App/AppEffects.h]: This would just be wrapper for the fastled one, just more convenient to not pass the same two params around

**Comment by ewowi** [comment]
Merged latest main into audio and did some post merge changes.
Expected a not of this in this PR but apparently not ...

File: src/App/AppEffects.h [hunk 4]
```diff
@@ -614,7 +614,7 @@ class AudioRings:public RingEffect {
 
 
     void loop() {
-      for (int i = 0; i < 7; i++) {
+      for (int i = 0; i < 7; i++) { // 7 rings
```
Comment (netmindz) [inline src/App/AppEffects.h:617]: Actually 7 EQ bands as was written for MSGEQ7 first

**Pull request merged**